File: user/src/main/java/com/kjj/user/repository/user/UserPolicyRepository.java
package com.kjj.user.repository.user;

import com.kjj.user.entity.UserPolicy;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;


public interface UserPolicyRepository extends JpaRepository<UserPolicy, Long> {

    @Query("UPDATE UserPolicy up SET up.defaultMenu = null WHERE up.defaultMenu = :menuId")
    void clearPolicyByDeletedMenu(Long menuId);
}

File: user/src/main/kotlin/com/kjj/user/repository/user/UserPolicyRepository.kt
package com.kjj.user.repository.user

import com.kjj.user.entity.UserPolicy
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query

interface UserPolicyRepository: JpaRepository<UserPolicy, Long> {
    @Query("update UserPolicy up set up.defaultMenu = null where up.defaultMenu = :menuId")
    fun clearPolicyByDeletedMenu(menuId: Long)
}